controller/src/beerocks/master/db/db.cpp

```diff
@@ -2299,16 +2302,14 @@ bool db::set_channel_scan_pool(const sMacAddr &mac, const std::unordered_set<uin
     }
 
     //Validate new dcs channel pool with hostap_supported_channels list
-    //This is a workaround
-    static uint8_t supported_channels_array[] = {
-        1,  2,   3,   4,   5,   6,   7,   8,   9,   10,  11,  36,  40,  44,  48,  52,  56,  60,
-        64, 100, 104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144, 149, 153, 157, 161, 165};
-    static size_t supported_channels_array_size =
-        sizeof(supported_channels_array) / sizeof(uint8_t);
-    std::set<uint8_t> supported_channels(supported_channels_array,
-                                         supported_channels_array + supported_channels_array_size);
+    auto supported_channels = get_hostap_supported_channels(network_utils::mac_to_string(mac));
     for (auto channel : channel_pool) {
-        if (supported_channels.find(channel) == supported_channels.end()) {
+        auto invalid_channel =
+            std::find_if(supported_channels.begin(), supported_channels.end(),
+                             [&channel](const beerocks::message::sWifiChannel &supported_channel) {
+                                 return supported_channel.channel == channel;
+                             });
+        if (invalid_channel == supported_channels.end()) {
```

Comment:
please use separate method instead  `is_channel_scan_pool_supported()` 
also, no need to scan all pool , just  return false for the first channel that is not supported

Reply:
added `is_channel_scan_pool_supported()`